src/macros.jl

```diff
@@ -195,13 +195,13 @@ moved to the set with `MOIU.shift_constant`.
 """
 function operator_to_set end
 
-operator_to_set(::Function, ::Union{Val{:(<=)},Val{:(≤)}}) = MOI.LessThan(0.0)
+operator_to_set(::Function, ::Union{Val{:(<=)},Val{:(≤)}}) = MOI.LessThan(false)
```

Comment:
Ouch

Reply:
It's similar to what `LinearAlgebra` is doing with `UniformScaling`. Using `Bool` at first and them promoting is a smooth path.
Whenever an expression sees a variable, it automatically promotes to `value_type` though.
So `x + 1` is still `AffExpr`, it does not build `GenericAffExpr{Int}` since otherwise we could loose a lot of time converting and hence allocating.